build/pmm-client-docker/entrypoint.sh

```diff
@@ -35,54 +35,143 @@ wait_for_port() {
-    --server-insecure-ssl \
-    --bind-address "${SRC_ADDR}" \
-    --client-address "${SRC_ADDR}" \
-    --client-name "${CLIENT_NAME}" \
-    ${ARGS}
-
-if [ -n "${DB_HOST}" ]; then
-    DB_ARGS+=" --host ${DB_HOST}"
-fi
-if [ -n "${DB_USER}" ]; then
-    DB_ARGS+=" --user ${DB_USER}"
-fi
-if [ -n "${DB_PASSWORD}" ]; then
-    DB_ARGS+=" --password ${DB_PASSWORD}"
-fi
-if [ -n "${DB_PORT}" ]; then
-    DB_ARGS+=" --port ${DB_PORT}"
-fi
-
-if [ -n "${DB_HOST}" -a "${DB_PORT}" ]; then
-    wait_for_port "${DB_HOST}" "${DB_PORT}"
-fi
-
-if [ -n "${DB_TYPE}" ]; then
-    pmm-admin add "${DB_TYPE}" \
+    --config-file=${AGENT_CONFIG_FILE} \
+    --server-address=${PMM_SERVER} \
+    --server-insecure-tls \
+    --container-id=${HOSTNAME} \
+    --container-name=${CLIENT_NAME} \
+    --ports-min=${CLIENT_PORT_MIN:-30100} \
+    --ports-max=${CLIENT_PORT_MAX:-30200} \
+    --listen-port=${CLIENT_PORT_LISTEN:-7777} \
+    ${ARGS} \
+    ${CLIENT_ADDR:-$SRC_ADDR} container ${CLIENT_NAME}
+
+    if [ -n "${DB_USER}" ]; then
+        DB_ARGS+=" --username=${DB_USER}"
+    fi
+    if [ -n "${DB_PASSWORD}" ]; then
+        DB_ARGS+=" --password=${DB_PASSWORD}"
+    fi
+    if [ -n "${DB_CLUSTER}" ]; then
+        DB_ARGS+=" --cluster=${DB_CLUSTER}"
+    fi
+
+    if [ -n "${DB_HOST}" -a "${DB_PORT}" ]; then
+        wait_for_port "${DB_HOST}" "${DB_PORT}"
+    fi
+
+    cat ${AGENT_CONFIG_FILE}
+
+    pmm-agent --config-file=${AGENT_CONFIG_FILE} \
+    --ports-min=${CLIENT_PORT_MIN:-30100} \
+    --ports-max=${CLIENT_PORT_MAX:-30200} > /usr/local/percona/pmm2/pmm-agent-tmp.log 2>&1 &
+
+    wait_for_url "http://127.0.0.1:7777"
+
+    cat /usr/local/percona/pmm2/pmm-agent-tmp.log
+
+    if [ -n "${DB_TYPE}" ]; then
+        case "${DB_TYPE}" in
+            mongodb )
+                pmm-admin add "${DB_TYPE}" \
+                            --skip-connection-check \
+                            --username "${MONGODB_USER}" \
+                            --password "${MONGODB_PASSWORD}" \
+                            --replication-set "${DB_REPLSET}"\
+                            --server-url="https://${PMM_USER}:${PMM_PASSWORD}@${PMM_SERVER}/" \
+                            --server-insecure-tls \
+                            --cluster "${DB_CLUSTER}" \
+                            --use-profiler \
+                            "${CLIENT_NAME}":27017 \
+
+                ;;
+            * )
+                pmm-admin add "${DB_TYPE}" \
+                            --skip-connection-check \
+                            --server-url="https://${PMM_USER}:${PMM_PASSWORD}@${PMM_SERVER}/" \
+                            --server-insecure-tls \
+                            ${DB_ARGS} \
+                            ${DB_HOST}:${DB_PORT}
+                ;;
+        esac
+    fi
+
+    kill %1
+    exec pmm-agent --config-file=/usr/local/percona/pmm2/config/pmm-agent.yaml --ports-min=${CLIENT_PORT_MIN:-30100} --ports-max=${CLIENT_PORT_MAX:-30200}
```

Comment:
minor style improvement
```suggestion
    exec pmm-agent \
        --config-file=/usr/local/percona/pmm2/config/pmm-agent.yaml \
        --ports-min=${CLIENT_PORT_MIN:-30100} \
        --ports-max=${CLIENT_PORT_MAX:-30200}
```

Reply:
done